src/System.Numerics.Tensors/src/System/Numerics/Tensors/ArrayUtilities.cs

```diff
@@ -0,0 +1,217 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace System.Numerics.Tensors
+{
+    internal static class ArrayUtilities
+    {
+        public const int StackallocMax = 16;
+
+        public static long GetProduct(ReadOnlySpan<int> dimensions, int startIndex = 0)
```

Comment:
Do we actually need the startIndex, or can this be done with slicing?  It's not a big deal as an internal API, but elsewhere even for internal methods we've opted to use slicing with spans rather than passing around offsets and counts as well.